backend/src/test/java/com/bikes/backend/repository/MongoUserRepositoryTest.java

```diff
@@ -0,0 +1,89 @@
+package com.bikes.backend.repository;
+
+import com.bikes.backend.model.MongoUser;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.springframework.test.annotation.DirtiesContext;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DataMongoTest
+class MongoUserRepositoryTest {
+
+	MongoUser testUser;
+	@Autowired
+	MongoUserRepository repository;
+
+
+	@BeforeEach
+	public void setUp() {
+		testUser = new MongoUser("Test ID", "Test username", "Test password", "BASIC");
+	}
+
+
+	@Nested
+	@DisplayName("findByUsername()")
+	class testFindById {
+
+		@Test
+		@DirtiesContext
+		@DisplayName("...should return the user if the user exists")
+		void findByUsername_returnsUserIfUserExists() {
+			// GIVEN
+			MongoUser expected = testUser;
+			repository.save(testUser);
+			// WHEN
+			MongoUser actual = repository.findByUsername(testUser.username()).orElseThrow();
```

Comment:
Here you are actually testing two methods: `.save` and `.findByUsername` - something to better avoid as we were told :)